codalab/common.py

```diff
@@ -272,33 +288,33 @@ def _get_signed_url(self, path, **kwargs):
             "/", 1
         )
         bucket = client.get_bucket(bucket_name)
-        blob = bucket.get_blob(blob_name)
+        blob = bucket.blob(blob_name)
         signed_url = blob.generate_signed_url(
             version="v4",
             expiration=datetime.timedelta(hours=1),
-            method="GET",
-            response_disposition=kwargs["content_disposition"],
-            response_type=kwargs["content_type"],
+            method=kwargs.get("method", "GET"),  # HTTP method. eg, GET, PUT
+            content_type=kwargs.get("request_content_type", None),
+            response_disposition=kwargs.get("content_disposition", None),
+            response_type=kwargs.get("content_type", None),
         )
         return signed_url
 
     def bundle_path_sas_url(self, **kwargs):
-        return self._get_sas_url(self.bundle_path, **kwargs)
-
-    def index_path_sas_url(self, **kwargs):
-        return self._get_sas_url(self.index_path, **kwargs)
-
-    def bundle_path_signed_url(self, **kwargs):
-        return self._get_signed_url(self.bundle_path, **kwargs)
-
-    def bundle_path_download_url(self, **kwargs):
         if self.storage_type == StorageType.AZURE_BLOB_STORAGE.value:
             return self._get_sas_url(self.bundle_path, **kwargs)
         elif self.storage_type == StorageType.GCS_STORAGE.value:
             return self._get_signed_url(self.bundle_path, **kwargs)
         else:
             raise UsageError(f"Does not support current storage type: {self.storage_type}")
 
+    def index_path_sas_url(self, **kwargs):
+        if self.storage_type == StorageType.AZURE_BLOB_STORAGE.value:
+            return self._get_sas_url(self.index_path, **kwargs)
+        elif self.storage_type == StorageType.GCS_STORAGE.value:
```

Comment:
This PR doesn't include GCS -- it's only for azure blob... can you remove this from this PR?

Reply:
This two lines are not specific for bypassing server upload, It is also used to bypass server download. I thinks we can live these two lines here.

Reply:
Thanks